sale_milestone_profile_invoicing/models/sale_order_line.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    amount_delivered_from_task = fields.Monetary(
+        string='Delivered from task',
+        compute='_compute_amount_delivered_from_task',
+        currency_field='currency_id',
+    )
+    amount_invoiced_from_task = fields.Monetary(
+        string='Invoiced from task',
+        compute='_compute_amount_invoiced_from_task',
+        currency_field='currency_id',
+    )
+    amount_delivered_from_task_company_currency = fields.Monetary(
+        string='Delivered from task (company currency)',
+        compute='_compute_amount_delivered_from_task',
+        currency_field='company_currency_id',
+    )
+    amount_invoiced_from_task_company_currency = fields.Monetary(
+        string='Invoiced from task (company currency)',
+        compute='_compute_amount_invoiced_from_task',
+        currency_field='company_currency_id',
+    )
+    company_currency_id = fields.Many2one(
+        related='company_id.currency_id', store=True
```

Comment:
make this readonly, and change the label to avoid a warning on runbot
```suggestion
        related='company_id.currency_id', store=True, readonly=True, string="Company currency"
```

Reply:
Done. A rebased, squash the all thing.